coprocessor/fhevm-engine/host-listener/src/database/tfhe_event_propagate.rs

```diff
@@ -363,6 +365,17 @@ impl Database {
         let insert_computation_bytes = |tx, result, dependencies_handles, dependencies_bytes, scalar_byte| {
             self.insert_computation_bytes(tx, tenant_id, result, dependencies_handles, dependencies_bytes, fhe_operation, scalar_byte, log)
         };
+
+        let _t = telemetry::tracer(
+            "handle_tfhe_event",
+            &log.transaction_hash.map(|h| h.to_vec()),
+        );
+
+        self.record_transaction_begin(
```

Comment:
should be done at the block level and not the event level. Ie in db_insert_block_no_retry once for each transaction_hash
(could be bulk insert later as improvement)

Reply:
There is a particular reason to call it inside `insert_tfhe_event`.

I wanted to make ensure that the stress-test tool that calls `insert_tfhe_event` also registers the transaction.

If there is any perf impact, we can refactor this into a batch insert.